components/admin_console/team_channel_settings/channel/details/channel_moderation.tsx

```diff
@@ -0,0 +1,347 @@
+    render = (): JSX.Element => {
+        const createPostsObject = this.props.channelPermissions && this.props.channelPermissions!.filter((permission) => permission.name === 'create_post')[0];
+        return (
+            <AdminPanel
+                id='channel_moderation'
+                titleId={t('admin.channel_settings.channel_moderation.title')}
+                titleDefault='Channel Moderation'
+                subtitleId={t('admin.channel_settings.channel_moderation.subtitle')}
+                subtitleDefault='Manage the actions available to channel members and guests.'
+            >
+                <div className='channel-moderation'>
+                    <div className='channel-moderation--body'>
+
+                        <table
+                            id='channel_moderation_table'
+                            className='channel-moderation--table'
+                        >
+                            <thead>
+                                <tr>
+                                    <th>
+                                        <FormattedMessage
+                                            id='admin.channel_settings.channel_moderation.permissions'
+                                            defaultMessage='Permissions'
+                                        />
+                                    </th>
+                                    <th>
+                                        <FormattedMessage
+                                            id='admin.channel_settings.channel_moderation.guests'
+                                            defaultMessage='Guests'
+                                        />
+                                    </th>
+                                    <th>
+                                        <FormattedMessage
+                                            id='admin.channel_settings.channel_moderation.members'
+                                            defaultMessage='Members'
+                                        />
+                                    </th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                                {this.props.channelPermissions && this.props.channelPermissions.map((entry) => {
+                                    return (
+                                        <ChannelModerationTableRow
+                                            key={entry.name}
+                                            name={entry.name}
+                                            guests={entry.roles.guests}
+                                            members={entry.roles.members}
+                                            onClick={this.props.onChannelPermissionsChanged}
+                                            teamScheme={this.props.teamScheme}
+                                            createPostsObject={entry.name === 'use_channel_mentions' ? createPostsObject : null}
```

Comment:
It would be nice if instead of this `createPostsObject` there were some generic way to declare dependencies between checkboxes, either through the value of another checkboxes directly or through a function. Perhaps a `disableIf` function that accepts the state and/or props as a parameter. It would be cleaner if a `<ChannelModerationTableRow>` had no knowledge of any specific permission.

Reply:
As we discussed, I have simplified and dumbed down the row so it doesn't contain logic.